### ISSUE TITLE
Meta(gt=...) doesn't work completely

### ISSUE THREAD

**Issue by aprilahijriyan** [jcrist/msgspec#334]
### Description

Here is the code to reproduce the bug

```python
from typing import Annotated
from msgspec import Meta

# working as expected
t = Annotated[int, Meta(gt=0)]
json.decode("0", type=t)

# Doesn't work
t = Annotated[int, Meta(gt=0, lt=10)]
print(json.decode("0", type=t)) # <<< This should be wrong right? because we expect the value to be greater than 0.
```

Tested on version:

* python: `3.9.11`
* msgspec: `0.12.0`

**Comment by aprilahijriyan** [comment]
I also tested on the latest version (0.13.1), still not working.

**Comment by jcrist** [comment]
Thanks for the excellent reproducible bug report, I can confirm that this is indeed a bug in the existing code. I'll try to get a fix for it up later.

**Comment by jcrist** [comment]
This is fixed by #335.

The bug only appears when mixing an integer `lt` constraint with `gt`/`ge`. A workaround you can use for now is to use `le` instead of `lt`:

```python
In [1]: from msgspec import Meta, json

In [2]: from typing import Annotated

In [3]: t = Annotated[int, Meta(gt=0, le=11)]  # use le instead

In [4]: json.decode("0", type=t)
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[4], line 1
----> 1 json.decode("0", type=t)

ValidationError: Expected `int` >= 1
```

Thanks again for the issue report!
